snap/snapcraft.yaml

```diff
@@ -0,0 +1,73 @@
+name: wezterm # you probably want to 'snapcraft register <name>'
+base: core22 # the base snap is the execution environment for this snap
+adopt-info: wezterm
+
+grade: stable
+confinement: classic
+
+apps:
+  wezterm:
+    command: bin/wezterm
+    common-id: org.wezfurlong.wezterm.desktop
+    desktop: assets/wezterm.desktop
+
+parts:
+  rust-deps:
+    plugin: nil
+    build-packages:
+      - curl
+    override-pull: |
+      curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+  wezterm:
+    plugin: rust
+    source: .
+    parse-info: [assets/wezterm.appdata.xml]
+    build-packages:
+      - libegl1-mesa-dev
+      - g++
+      - libwayland-dev
+      - libssl-dev
+      - libfontconfig-dev
+      - libx11-xcb-dev
+      - libxcb-ewmh-dev
+      - libxcb-icccm4-dev
+      - libxcb-image0-dev
+      - libxcb-keysyms1-dev
+      - libxcb-randr0-dev
+      - libxcb-render0-dev
+      - libxcb-util-dev
+      - libxcb-xkb-dev
+      - libxkbcommon-dev
+      - libxkbcommon-x11-dev
+      - python3
+      - pkg-config
+    after:
+      - rust-deps
+    override-pull: |
+      craftctl default
+      craftctl set version=$(git describe --tags | head -c 32)
+    stage-packages:
+      - libx11-6
+      - libx11-xcb1
+      - libxau6
+      - libxdmcp6
+      - libfontconfig1
+      - libwayland-client0
+      - libwayland-egl1
+      - libxcb-image0
+      - libxcb-shm0
+      - libxcb-util1
+      - libxcb-xkb1
+      - libxcb1
+      - libxkbcommon-x11-0
+      - libxkbcommon0
+    override-build: |
+      cargo build --release --features distro-defaults
+      mkdir -p $SNAPCRAFT_PART_INSTALL/bin
+      for component in wezterm wezterm-gui wezterm-mux-server strip-ansi-escapes ; do
```

Comment:
I've paged out pretty much everything I know about snaps... how do users explicitly launch one of these binaries, vs. just launching wezterm itself?

Sort of related; is there a reasonable way to export the shell integration script(s) (https://github.com/wez/wezterm/tree/main/assets/shell-integration) from the snap?  There isn't a way to do this with the AppImage and I doubt that is possible for snap, I'm just asking in case there is!

Reply:
I believe there *is* some support for exporting shell completions from snaps.

Actually, now I think about it, because this will be a `classic` snap, we can just have an `install` hook copy the integration into the right place. Does the release machinery know where ”the right place” is, so I can copy it?

Reply:
this is where we build the .deb package and copy the shell integration and completion files; the shell integration lands in `/etc/profile.d`, but the destination for shell completion varies by shell:

https://github.com/wez/wezterm/blob/6b21e6ffde46d1345fdfc5a1437054fa8d964e7c/ci/deploy.sh#L248-L250

Note that on rpm based systems the shell completions live in completely different locations:
https://github.com/wez/wezterm/blob/6b21e6ffde46d1345fdfc5a1437054fa8d964e7c/ci/deploy.sh#L181-L182

Reply:
> I've paged out pretty much everything I know about snaps... how do users explicitly launch one of these binaries, vs. just launching wezterm itself?

Any binaries that want to end up in `$PATH` need to be explicitly added to the `apps:` section, above. Anything invoked by the `wezterm` binary itself is fine, though, as you've already got code to locate the relative path to these (as far as I can tell - everything *seems* to work :grin:).

I assumed that users didn't need to directly invoke the other binaries, but maybe actually we should expose `wezterm-gui` (with the `.desktop` file) *and* `wezterm` (without the `.desktop` file?), and maybe also `wezterm-mux-server`?

Reply:
Yeah, I think exposing/exporting all the binaries to the PATH makes sense, as there are some folks that want to use the mux server directly on remote hosts.

Only wezterm-gui needs to have a `.desktop` file.